tap_zendesk/__init__.py

```diff
@@ -117,16 +126,43 @@ def do_sync(client, catalog, state, start_date):
     singer.write_state(state)
     LOGGER.info("Finished sync")
 
+def oauth_auth():
+    try:
+        oauth_args = singer.utils.parse_args(OAUTH_CONFIG_KEYS)
+        creds = {
+            "subdomain": args['subdomain'],
+            "oauth_token": oauth_args.config['access_token'],
+        }
+
+        client = Zenpy(**creds)
+        LOGGER.info("OAuth authentication successful.")
+        return client
+    except ZenpyException:
+        LOGGER.error("OAuth authentication failed.")
+    except Exception:
```

Comment:
Is this a `ValueError` or other kind of specific exception? Could we not catch `Exception` here by just checking if the args contain a particular key?

Reply:
The `parse_args` will raise an `Exception` (see https://github.com/singer-io/singer-python/blob/dfad8fc467704900e0f268a05919bd34976aaa0e/singer/utils.py#L150) if the `required_keys` are not present.

So we can't use `parse_args` to validate that the config is there without catching an `Exception` when the config is not there.

I'll do the check manually then.